core/src/main/java/de/danoeh/antennapod/core/feed/LocalFeedUpdater.java

```diff
@@ -0,0 +1,137 @@
+package de.danoeh.antennapod.core.feed;
+
+import android.content.Context;
+import android.media.MediaMetadataRetriever;
+import android.net.Uri;
+import android.webkit.MimeTypeMap;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import de.danoeh.antennapod.core.storage.DBTasks;
+
+public class LocalFeedUpdater {
+
+    private static long getFileDuration(File f) {
+        MediaMetadataRetriever mediaMetadataRetriever = new MediaMetadataRetriever();
+        mediaMetadataRetriever.setDataSource(f.getAbsolutePath());
+        String durationStr = mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION);
+        return Long.parseLong(durationStr);
+    }
+
+    /** Starts the import process. */
+    public static void startImport(Uri uri, Context context) {
+        File f = new File(uri.getPath());
+        if (!f.isDirectory()) {
+            throw new RuntimeException("invalid path");
+        } else {
+            startImportDirectory(uri, context);
+        }
+    }
+
+    private static Feed startImportDirectory(Uri uri, Context context) {
+        //create a feed object for this directory
+        File f = new File(uri.getPath());
+        String dirUrl = uri.toString();
+        Feed dirFeed = new Feed(dirUrl, null, dirUrl);
+        dirFeed.setItems(new ArrayList<>()); //this seems useless but prevents an exception
+        //find the feed for this directory (if it exists), or create one
+        dirFeed = DBTasks.updateFeed(context, dirFeed)[0];
+        return dirFeed;
+    }
+
+    public static void updateFeed(Feed feed, Context context) {
+        //make sure it is the latest 'version' of this feed from the db
+        //NOTE that, if the caller is always DBTasks, this is unnecessary; but maybe it comes from
+        //somewhere else in the future?
+        // Feed dirFeed = DBTasks.updateFeed(context, feed)[0];
+        Feed dirFeed = feed;
+
+        String uriStr = dirFeed.getDownload_url();
+        File f = new File(uriStr.substring("file:".length())); //ugly
```

Comment:
The original code had URIs, so I just kept it.
I think it is best to avoid passing strings around, so one way here would be to use File objects instead of URIs.

Reply:
Latest commit does that. Seems to work fine.